src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,166 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                userOrders.add(getOrderFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get orders for user id " + userId, e);
+        }
+        for (Order order : userOrders) {
+            order.setProducts(getOrdersProducts(order.getId()));
+        }
+        return userOrders;
+    }
+
+    @Override
+    public Order create(Order order) {
+        String query = "INSERT INTO orders(user_id) VALUES (?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, order.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            while (resultSet.next()) {
+                order.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create order " + order.toString(), e);
+        }
+        createOrdersProducts(order.getId(), order.getProducts());
+        return order;
+    }
+
+    @Override
+    public Optional<Order> get(Long id) {
+        Order order = new Order();
+        String query = "SELECT order_id, user_id FROM orders WHERE deleted = FALSE "
+                + "AND order_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                order = getOrderFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get shopping cart with id " + id, e);
+        }
+        order.setProducts(getOrdersProducts(id));
+        return Optional.of(order);
+    }
+
+    @Override
+    public Order update(Order order) {
+        String query = "UPDATE orders_products SET deleted = TRUE WHERE order_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, order.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update order with id " + order.getId(), e);
+        }
+        createOrdersProducts(order.getId(), order.getProducts());
```

Comment:
what about removing old relations before creating new?

Reply:
old relations removed with sql query on line 83

Reply:
added separate method for this purpose - deleteOrdersProducts()